crates/oxc_linter/src/rules/eslint/no_control_regex.rs

```diff
@@ -0,0 +1,351 @@
+///
+/// e.g.:
+/// * /foo/ -> "foo"
+/// * new RegExp("foo") -> foo
+///
+/// note: [`RegExpFlags`] and [`Span`]s are both tiny and cloneable.
+fn regex_pattern<'a>(node: &AstNode<'a>) -> Option<(&'a Atom, Option<RegExpFlags>, Span)> {
+    let kind = node.kind();
+    match kind {
+        // regex literal
+        AstKind::RegExpLiteral(reg) => Some((&reg.regex.pattern, Some(reg.regex.flags), reg.span)),
+
+        // FIXME: we need a more graceful way to handle NewExpr/CallExprs
+
+        // new RegExp()
+        AstKind::NewExpression(expr) => {
+            // constructor is RegExp,
+            if let Expression::Identifier(ident) = &expr.callee &&
+            ident.name == "RegExp" &&
+            // which is provided at least 1 parameter,
+            expr.arguments.len() > 0 &&
+            // where the first one is a string literal
+            // note: improvements required for strings used via identifier
+            // references
+            let Argument::Expression(Expression::StringLiteral(pattern)) = &expr.arguments[0]
+            {
+                // get pattern from arguments. Missing or non-string arguments
+                // will be runtime errors, but are not covered by this rule.
+                // Note that we're intentionally reporting the entire "new
+                // RegExp("pat") expression, not just "pat".
+                Some((&pattern.value, extract_flags(&expr.arguments), kind.span()))
+            } else {
+                None
+            }
+        }
+
+        // RegExp()
+        AstKind::CallExpression(expr) => {
+            // constructor is RegExp,
+            if let Expression::Identifier(ident) = &expr.callee &&
+            ident.name == "RegExp" &&
+            // which is provided at least 1 parameter,
+            expr.arguments.len() > 0 &&
+            // where the first one is a string literal
+            // note: improvements required for strings used via identifier
+            // references
+            let Argument::Expression(Expression::StringLiteral(pattern)) = &expr.arguments[0]
+            {
+                // get pattern from arguments. Missing or non-string arguments
+                // will be runtime errors, but are not covered by this rule.
+                // Note that we're intentionally reporting the entire "new
+                // RegExp("pat") expression, not just "pat".
+                Some((&pattern.value, extract_flags(&expr.arguments), kind.span()))
+            } else {
+                None
+            }
+        }
+        _ => None,
+    }
+}
+
+fn control_patterns(pattern: &Atom) -> Matches<'static, '_> {
+    lazy_static! {
+        static ref CTL_PAT: Regex = Regex::new(
+            r"([\x00-\x1f]|(?:\\x\w{2})|(?:\\u\w{4})|(?:\\u\{\w{1,4}\}))"
+            // r"((?:\\x\w{2}))"
+        ).unwrap();
+    }
+    CTL_PAT.find_iter(pattern.as_str())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    macro_rules! test_vec {
```

Comment:
you can use `Tester::new_without_config`

Reply:
I try to avoid macros and just expand out the code, personally I find it easier to navigate without macros (due to IDE)

Reply:
Fixed